source/common/http/utility.cc

```diff
@@ -136,17 +136,27 @@ bool Utility::isInternalRequest(const HeaderMap& headers) {
   return Network::Utility::isInternalAddress(forwarded_for->value().c_str());
 }
 
-uint64_t Utility::parseCodecOptions(const Json::Object& config) {
-  uint64_t ret = 0;
+Http2Settings Utility::parseHttp2Settings(const Json::Object& config) {
+  Http2Settings ret;
+
   std::string options = config.getString("http_codec_options", "");
   for (const std::string& option : StringUtil::split(options, ',')) {
     if (option == "no_compression") {
```

Comment:
This feature should be marked `DEPRECATED` in comments. Also we should track this in #1012.